wallet/tests/data_requests.rs

```diff
@@ -154,3 +155,55 @@ fn test_data_request_report_json_serialization() {
         report.tally.partial_results.unwrap().last().unwrap()
     );
 }
+
+/// This is a rather end-2-end test that use a RNG input and checks whether the
+/// final `RadonReport` complies with the Witnet Wallet API.
+#[test]
+fn test_try_data_request_rng() {
```

Comment:
This test does not fail if I remove the other changes of this pull request.

You can add the following line:

```rust
    assert_eq!(report.tally.result, RadonTypes::Bytes(RadonBytes::from(vec![254, 69, 239, 98, 10, 203, 40, 254, 158, 109, 130, 121, 108, 158, 200, 110, 49, 9, 152, 250, 37, 113, 2, 99, 125, 65, 249, 61, 118, 176, 184, 132])));
```

Also, this test is in the `wallet` crate, which does not make much sense. Let's move it to somewhere inside the `rad` crate. And remove all the partial_results asserts, because they are already checked in another test.